src/stdune/path.mli

```diff
@@ -63,6 +84,7 @@ module Build : sig
   val append_source : t -> Source.t -> t
 
   val append_local : t -> Local.t -> t
+  val append_relative : t -> Relative.t -> t
```

Comment:
What do we do about attempts to go outside the root here?

Reply:
Interesting. I guess we can either error out or just return `Path.External.root`. I'm not certain.

Reply:
You clearly(?) don't want "../foo" go to "foo" if done at the root. So if we make that fail, we should probably also make plain ".." fail.

Reply:
(actually, ../foo does go to foo at the unix root, so that's not unheard of)

Reply:
> You clearly(?) don't want "../foo" go to "foo" if done at the root. So if we make that fail, we should probably also make plain ".." fail.

Do you mean `External.append_relative External.root (Relative.of_string "../")`?

What is wrong with this working just like `Filename.concat`? E.g. `/../usr` is a valid path.

Reply:
Yeah, for `External` I think it's fine to just defer to OS semantics. I was more thinking about the in-tree paths.